### ISSUE TITLE
Stats visitor count discrepency

### ISSUE THREAD

**Issue by nbradbury** [woocommerce/woocommerce-android#1805]
If I view stats for this month (January), I see 3 visitors.

![1](https://user-images.githubusercontent.com/3903757/72075722-50de1f80-32c2-11ea-8a0c-1fe479facf0c.png)

But if I view stats for this year, I see 13 visitors.

![2](https://user-images.githubusercontent.com/3903757/72075760-5f2c3b80-32c2-11ea-93b7-006a0114055a.png)

And if I click the January bar, I see 0 visitors.

![3](https://user-images.githubusercontent.com/3903757/72075779-65bab300-32c2-11ea-976c-75ad4a1abe95.png)

These visitor counts should all be the same. Note that Calypso shows 3 visitors.

<img width="376" alt="Screen Shot 2020-01-09 at 9 22 35 AM" src="https://user-images.githubusercontent.com/3903757/72075879-94388e00-32c2-11ea-8bd4-ceb7bf2abce3.png">

**Comment by anitaa1990** [comment]
Debugging this further: From an earlier discussion, it was found that we would need to pass the `date` field in `YYYY-MM-dd` format when fetching visitor stats. But this is not the case when we fetch visitor stats data for `StatsGranularity.YEARS`. I added a fix in this [patch](https://github.com/wordpress-mobile/WordPress-FluxC-Android/commit/7c77c30df7160bf51cc48227712c958b369b2552) and this looks like it solves the issue.

But now, I see that when fetching `MONTHS` tab, the visitor stats is 5 but when fetching `YEARS` tab, the visitor stats is only 4. The visitor count for the date `2019-01-02` is not counted when fetching visitor stats. 

#### Response from `MONTHS` TAB:
```
{
    "date": "2020-01-09",
    "unit": "day",
    "fields": [
        "period",
        "visitors"
    ],
    "data": [
        [
            "2020-01-01",
            0
        ],
        [
            "2020-01-02",
            1
        ],
        [
            "2020-01-03",
            0
        ],
        [
            "2020-01-04",
            1
        ],
        [
            "2020-01-05",
            0
        ],
        [
            "2020-01-06",
            0
        ],
        [
            "2020-01-07",
            0
        ],
        [
            "2020-01-08",
            1
        ],
        [
            "2020-01-09",
            2
        ]
    ]
}
```

#### Response from `YEARS` TAB
```
{
    "date": "2020-01-09",
    "unit": "month",
    "fields": [
        "period",
        "visitors"
    ],
    "data": [
        [
            "2019-12-01",
            13
        ],
        [
            "2020-01-01",
            4
        ]
    ]
}
```

**Comment by AmandaRiu** [comment]
@anitaa1990 

> But now, I see that when fetching MONTHS tab, the visitor stats is 5 but when fetching YEARS tab, the visitor stats is only 4. The visitor count for the date 2019-01-02 is not counted when fetching visitor stats.

Does this change if you set the end date for the MONTHS call to tomorrow instead of today?

**Comment by AmandaRiu** [comment]
Wrote this up in a p2 post as well. See `p91TBi-2Pn-p2`.

**Comment by Garance91540** [comment]
Hi @AmandaRiu ! I answered on the P2 but I actually think unique visitors is a more important data point for a store owner... I could be wrong and we need to test this assumption with our users. More info on the P2

**Comment by erricgunawan** [comment]
I got a report on 3249785-zen that looks like it's the same issue here.

Or is that a different one?

**Comment by AmandaRiu** [comment]
@erricgunawan I looked over the issue ticket and this doesn't look to be the same thing since this issue specifically leads to users seeing _**higher**_ visitor counts, not lower. I'll also reply in the ticket.

**Comment by erricgunawan** [comment]
Thanks for looking @AmandaRiu !

**Comment by designsimply** [comment]
During WooCommerce Android 6.2 beta testing, @pmusolino observed that the number of visitors shown on the "This Year" screen for iOS vs Android can be off by one or a few visits. I think this might be caused by what Anita explained in her [comment above](https://github.com/woocommerce/woocommerce-android/issues/1805#issuecomment-572941402) 

When addressing this issue, would it be possible to also compare against WCiOS and verify that the visitor stats are counted in the same way? If you feel this should be address separately, can you create a new issue (or ask me to)?

Here is a quote from another discussion I am including for completeness: 

> @rachelmcr mentioned that we should make sure we’re getting the visitors count directly from the API, and not doing our own calculations, to make sure we’re consistent. The backend stats only counts unique visitors over the selected period, so if we calculate the visitor count ourselves we might not be only counting unique visitors. For example, consider if only 1 person visits my store but they visit every month. When I select “this month” it should be 1 visitor and when I select “this year” it should also be 1 visitor, since it’s the same visitor every month. But if we calculate “this year” by adding up the “this month” count each month, we’d show more visitors than the actual unique visitor count.

<sup>(internal reference: p5T066-21o-p2#comment-7514)</sup>

**Comment by anitaa1990** [comment]
@designsimply, I think this is fixed in #3706 in 6.3 release. I'm seeing similar stats on android and iOS for my test site but would like to get your confirmation before closing this one

**Comment by designsimply** [comment]
I tested today using 6.7-rc-2 and compared Visitors for Month vs Year > Month and the stats did not match in my test.

Here are the steps I used for testing: 

1. Go to wordpress.com/stats/month/ select the site you are testing and note the number of Visitors for the current month. 
1. Go to WCAndroid > My Store > This Month and note the number of Visitors for the current month.
1. Go to WCAndroid > My Store > This Year and tap the current month in the graph and note the number of Visitors.
1. Pull-to-refresh, re-select the current month, and re-check the number of Visitors.

**Result:** 

* Calypso Stats May 2021 = 2 Visitors
* WooCommerce Android May 2021 - This Month = 2 Visitors
* WooCommerce Android 2021 - This Year > tap current month = 0 Visitors

Calypso Month (2)|WCAndroid Month (2)|WCAndroid Year > Month (0)
---|---|---
![Screenshot_20210519-171847](https://user-images.githubusercontent.com/1119271/118897737-2180ef00-b8c8-11eb-9c8c-2d7e8d2b77f0.png)|![Screenshot_20210519-172846](https://user-images.githubusercontent.com/1119271/118897735-20e85880-b8c8-11eb-8b13-9b2cc300dbac.png)|![Screenshot_20210519-172840](https://user-images.githubusercontent.com/1119271/118897736-2180ef00-b8c8-11eb-81c6-9cd6301b36f0.png)

**Notes:** I used madefortesting011121.wpcomstatging.com which is a WordPress.com Business / Atomic site running WordPress 5.7.2 Jetpack 9.7 WooCommerce 5.3.0.
